plenum/test/node_catchup/test_node_catchup_with_connection_problem.py

```diff
@@ -0,0 +1,203 @@
+    this case catchup node has no quorum with f+1 CONSISTENCY_PROOFs for the
+    longer transactions list. It need to request CONSISTENCY_PROOFs again and
+    finishes catchup.
+    Test makes sure that the node eventually finishes catchup'''
+    node_to_disconnect = txnPoolNodeSet[-1]
+
+    def unpatch_after_call(proof, frm):
+        global call_count
+        call_count += 1
+        if call_count >= lost_count:
+            # unpatch processConsistencyProof after lost_count calls
+            monkeypatch.undo()
+            call_count = 0
+
+    sdk_send_random_and_check(looper, txnPoolNodeSet,
+                              sdk_pool_handle, sdk_wallet_steward, 5)
+
+    # restart node
+    disconnect_node_and_ensure_disconnected(looper,
+                                            txnPoolNodeSet,
+                                            node_to_disconnect)
+    looper.removeProdable(name=node_to_disconnect.name)
+    sdk_send_random_and_check(looper, txnPoolNodeSet,
+                              sdk_pool_handle, sdk_wallet_steward,
+                              2)
+
+    nodeHa, nodeCHa = HA(*node_to_disconnect.nodestack.ha), HA(
+        *node_to_disconnect.clientstack.ha)
+    config_helper = PNodeConfigHelper(node_to_disconnect.name, tconf,
+                                      chroot=tdir)
+    node_to_disconnect = TestNode(node_to_disconnect.name,
+                                  config_helper=config_helper,
+                                  config=tconf,
+                                  ha=nodeHa, cliha=nodeCHa,
+                                  pluginPaths=allPluginsPath)
+    # patch processConsistencyProof
+    monkeypatch.setattr(node_to_disconnect.ledgerManager,
+                        'processConsistencyProof',
+                        unpatch_after_call)
+    # add node_to_disconnect to pool
+    looper.add(node_to_disconnect)
+    txnPoolNodeSet[-1] = node_to_disconnect
+    looper.run(checkNodesConnected(txnPoolNodeSet))
+    waitNodeDataEquality(looper, node_to_disconnect, *txnPoolNodeSet)
+
+
+def test_catchup_with_lost_last_consistency_proof(txnPoolNodeSet,
+                                                  looper,
+                                                  sdk_pool_handle,
+                                                  sdk_wallet_steward,
+                                                  tconf,
+                                                  tdir,
+                                                  allPluginsPath,
+                                                  monkeypatch,
+                                                  lost_count):
+    '''Skip processing of lost_count CONSISTENCY_PROOFs that resieve for
+    consistency proof request (not for ledger status). In this case catchup node
+    has no quorum for proofing some size of transactions list. It need to request
+    CONSISTENCY_PROOFs again and finishes catchup. Test makes sure that the node
+    eventually finishes catchup'''
```

Comment:
Currently this test does not verify what it must because all the ConsistencyProofs being received by the restarted node are the same. So when the restarted node gathers initial f+1 ConsistencyProofs, it starts catch-up, because these f+1 messages are the same.

Reply:
Removed this test and added new with check cancel of schedule with requesting ledger statuses and consistency proofs.